optimum/habana/transformers/generation/utils.py

```diff
@@ -2437,6 +2437,17 @@ def _sample(
         if streamer is not None:
             streamer.end()
 
+        if batch_size > 1 and has_eos_stopping_criteria:
+            eos_token_id = generation_config.eos_token_id
+            idx_bs = generation_config.max_length
+            for i in range(batch_size):
+                for idx in range(len(input_ids[i])):
+                    if input_ids[i][idx] == eos_token_id:
+                        idx_bs = idx
+                    if idx > idx_bs:
+                        input_ids[i][idx] = pad_token_id
+                idx_bs = generation_config.max_length
```

Comment:
do we still want to return padding token in this case or just stop here?

Reply:
according to customer requirement, we need to padding token after <|endoftext|>